Source/Client/Desyncs/DesyncReporter.cs

```diff
@@ -0,0 +1,204 @@
+                    desyncReport.AddEntry("desync_info", desyncInfo.ToString());
+
+                    zip.Save();
+
+                    //Add the basic info to the report
+                    desyncReport.AddEntry("info", zip["info"].GetBytes());
+
+                    desyncReport.AddEntry("ign", Multiplayer.session.GetPlayerInfo(Multiplayer.session.playerId).username);
+                    desyncReport.AddEntry("steamName", SteamUtility.SteamPersonaName);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error($"Exception writing desync info: {e}");
+            }
+        }
+
+        public static void Upload()
+        {
+            //Report desync to the server
+            var request = (HttpWebRequest) WebRequest.Create("http://multiplayer.samboycoding.me/api/desync/upload");
+//                    var request = (HttpWebRequest) WebRequest.Create("http://localhost:4193/api/desync/upload");
+            request.Method = "POST";
+            request.ContentType = "application/zip";
+
+            using (var stream = new MemoryStream())
+            {
+                desyncReport.Save(stream);
+
+                stream.Seek(0, SeekOrigin.Begin);
+                request.ContentLength = stream.Length;
+                var data = new byte[stream.Length];
+                stream.Read(data, 0, data.Length);
+
+                using (var outStream = request.GetRequestStream())
+                    outStream.Write(data, 0, data.Length);
+            }
+
+            //TODO: Some user interaction here?
+            try
+            {
+                using (var response = (HttpWebResponse) request.GetResponse())
+                {
+                    if (response.StatusCode != HttpStatusCode.OK)
+                    {
+                        Log.Error("Failed to report desync; Status code " + response.StatusCode);
+                    }
+                    else
+                    {
+                        using (var stream = response.GetResponseStream())
+                        using (var reader = new StreamReader(stream))
+                        {
+                            var desyncReportId = reader.ReadToEnd();
+                            Log.Message("Desync Reported with ID " + desyncReportId);
+                            window.reportId = desyncReportId;
+                        }
+                    }
+                }
+            }
+            catch (WebException e)
+            {
+                Log.Error("Failed to report desync; " + e.Message);
+                window.reportId = "";
+            }
+
+            window.reporting = false;
+        }
+
+        private static string GetDesyncStackTraces(ClientSyncOpinion local, ClientSyncOpinion remote, out int index)
+        {
+            return Multiplayer.game.sync.GetDesyncStackTraces(local, remote, out index);
+        }
+
+        private static string FindFileNameForNextDesyncFile()
```

Comment:
I believe this is being done every time the user desyncs? If so, wouldn't it be better to read the required data, like how many desync files are present, and what the current max desync number is, when a DesyncReporter is being instantiated? This would reduce the amount of disk reads.

Reply:
DesyncReporter is static, so it'd be a check at launch, and the amount of disk activity this generates is negligible next to e.g. an autosave or actually writing a desync.

Reply:
I'd also say that this method already existed, I just moved it to this class from SyncCoordinator.